File: go-backend/dto/ddns.go
package dto

import "encoding/json"

type DdnsProviderDto struct {
	Name       string          `json:"name" binding:"required"`
	Type       string          `json:"type" binding:"required"`
	Credential json.RawMessage `json:"credential" binding:"required"`
}

type DdnsProviderUpdateDto struct {
	ID         int64           `json:"id" binding:"required"`
	Name       string          `json:"name"`
	Credential json.RawMessage `json:"credential"`
}

type DdnsDomainDto struct {
	ProviderId  int64  `json:"providerId" binding:"required"`
	GroupId     int64  `json:"groupId" binding:"required"`
	Domain      string `json:"domain" binding:"required"`
	RecordName  string `json:"recordName"`
	RecordType  string `json:"recordType" binding:"required"`
	AutoResolve int    `json:"autoResolve"`
}

type DdnsDomainUpdateDto struct {
	ID          int64  `json:"id" binding:"required"`
	ProviderId  int64  `json:"providerId"`
	Domain      string `json:"domain"`
	RecordName  string `json:"recordName"`
	RecordType  string `json:"recordType"`
	AutoResolve *int   `json:"autoResolve"`
}

File: nextjs-frontend/lib/api/ddns.ts
import { post } from './client';

export type DdnsProvider = {
  id: number;
  name: string;
  type: string;
  createdTime: number;
  updatedTime: number;
};

export type DdnsDomain = {
  id: number;
  providerId: number;
  groupId: number;
  domain: string;
  recordName: string;
  recordType: string;
  autoResolve: number;
  currentRecord: string;
  currentMemberId: number;
  createdTime: number;
  updatedTime: number;
  status: number;
};

export type DdnsProviderCreatePayload = {
  name: string;
  type: string;
  credential: unknown;
};

export type DdnsProviderUpdatePayload = {
  id: number;
  name?: string;
  credential?: unknown;
};

export type DdnsDomainCreatePayload = {
  providerId: number;
  groupId: number;
  domain: string;
  recordName?: string;
  recordType: string;
  autoResolve: number;
};

export type DdnsDomainUpdatePayload = {
  id: number;
  providerId?: number;
  domain?: string;
  recordName?: string;
  recordType?: string;
  autoResolve?: number;
};

export const getDdnsProviderTypes = () => post<string[]>('/ddns/provider/types');
export const getDdnsProviderList = () => post<DdnsProvider[]>('/ddns/provider/list');
export const createDdnsProvider = (data: DdnsProviderCreatePayload) => post('/ddns/provider/create', data);
export const updateDdnsProvider = (data: DdnsProviderUpdatePayload) => post('/ddns/provider/update', data);
export const deleteDdnsProvider = (id: number) => post('/ddns/provider/delete', { id });

export const getDdnsDomainList = () => post<DdnsDomain[]>('/ddns/domain/list');
export const createDdnsDomain = (data: DdnsDomainCreatePayload) => post('/ddns/domain/create', data);
export const updateDdnsDomain = (data: DdnsDomainUpdatePayload) => post('/ddns/domain/update', data);
export const deleteDdnsDomain = (id: number) => post('/ddns/domain/delete', { id });
export const resolveDdnsDomain = (id: number) => post('/ddns/domain/resolve', { id });
